nautobot/core/views/__init__.py

```diff
@@ -220,3 +225,34 @@ def render_graphiql(self, request, **data):
         data["saved_graphiql_queries"] = GraphQLQuery.objects.all()
         data["form"] = GraphQLQueryForm
         return render(request, self.graphiql_template, data)
+
+
+class NautobotAppMetricsCollector(Collector):
+    """Custom Nautobot metrics collector.
+
+    Metric collector that reads from registry["plugin_metrics"] and yields any metrics registered there."""
+
+    def collect(self):
+        """Collect metrics from plugins."""
+        start = time.time()
+        for metric in registry["app_metrics"]:
+            yield from metric()
+        gauge = GaugeMetricFamily("nautobot_app_metrics_processing_ms", "Time in ms to generate the app metrics")
+        duration = time.time() - start
+        gauge.add_metric([], format(duration * 1000, ".5f"))
+        yield gauge
+
+
+def custom_app_metric_view(request):
+    """Exports /metrics.
+
+    This overwrites the default django_prometheus view to inject metrics from Nautobot apps."""
+    if "PROMETHEUS_MULTIPROC_DIR" in os.environ or "prometheus_multiproc_dir" in os.environ:
+        prometheus_registry = prometheus_client.CollectorRegistry()
+        multiprocess.MultiProcessCollector(prometheus_registry)
+    else:
+        prometheus_registry = prometheus_client.REGISTRY
+    nb_app_collector = NautobotAppMetricsCollector()
+    prometheus_registry.register(nb_app_collector)
```

Comment:
Let's see if there isn't a potentially better place to register this collector without needing to do it every call time.

Reply:
I could theoretically evaluate persisting the registry over multiple calls to the view, but I don't think this is a good idea in the end for two reasons:
1. This would have us deviate a lot further from the proven `django_prometheus` implementation of this view
2. The collector themselves are implemented as generators. When the metrics are generated, these generators are depleted. Thus, I would have to work around that, which feels hacky.

Reply:
I explained some of this in a comment as well.

Reply:
I think you raise a good point here. Just for my technical understanding - when would we be running in a non-multiprocess scenario - on a deployment directly to a VM?

Reply:
Win for the tests! They replicated this error you caught: https://github.com/nautobot/nautobot/actions/runs/4366588475/jobs/7636881567

Reply:
Fixed now.